confmap/confmap.go

```diff
@@ -189,6 +190,13 @@ func encoderConfig(rawVal interface{}) *encoder.EncoderConfig {
 	}
 }
 
+// case-sensitive version of the callback to be used in the MatchName property
```

Comment:
I was following the convention of the functions used in `DecodeHook`, including `expandNilStructPointersHookFunc`. But I guess this does not make sense since `caseSensitiveMatchName` is not part of that and is just a function with defined signature.
I will change.